.github/workflows/ci-build.yml

```diff
@@ -102,19 +100,19 @@ jobs:
           name: 'Unit Test Results'
           if-no-files-found: error
           path: |
-            **/build/reports/tests/*/
+            **/build/test-results/*/
 
-      - name: "Publish 'Unit Results' check suite."
+      - name: "Upload 'Unit Test Reports' artifact."
         if: ${{ success() || failure() }}
-        uses: EnricoMi/publish-unit-test-result-action@v2
+        uses: actions/upload-artifact@v4
         with:
-          check_name: '🔔 Test: Unit Results'
```

Comment:
Why wasn't this working? Why does it need to be in a separate workflow?

Reply:
`checks: write` is downgraded to `checks: read` in forks on `pull_request`, so your PRs would give a "Resource is inaccessible by integration" error. The action author hid that in a warning message, instead of "crashing":

https://github.com/TWiStErRob/net.twisterrob.ghlint/issues/286#issuecomment-2138329986